eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -0,0 +1,79 @@
+parameters:
+  PackageName: ""
+  EnvVars: []
+  MaxParallel: 0
+  Matrix:
+    Linux_Node10X:
+      OSName: "Linux"
```

Comment:
It feels like OSName and OSVmImage are redundant and we could eliminate one and potentially even set one based on the other if we still wanted needed them.

Also it seems like the JS repo is missing the verify os agent steps (https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/jobs/tests.yml#L42-L52) which we should consider including to ensure we run on the correct OS.

Reply:
The OSVmImage can imply the OSName just as the TestType can imply the TestResultsFiles. 

Also added in the OS test